test/python/sdpa/mxfp8.py

```
@@ -455,10 +458,299 @@ def generate_graph_bwd(b, h_q, h_k, h_v,
+            fill_sparse_small_int(q_f32, rng_data, sparsity=0.8, abs_max=2)
+        if s_kv_i:
+            fill_sparse_small_int(k_f32, rng_data, sparsity=0.8, abs_max=2)
+            fill_sparse_small_int(v_f32, rng_data, sparsity=0.8, abs_max=2)
+        if s_q_i and s_kv_i:
+            # keep a few q rows in the deeply-negative-score regime; must run
+            # before quantization (same contract as the dense path)
+            inject_negative_score_rows(q_f32, k_f32, rng_data, attn_scale=attn_scale)
+        q8, dqq, sfq = _quantize_seq(q_f32, h_q, s_q_i, d_qk, torch_itype, block_size, columnwise=False)
+        k8, dqk, sfk = _quantize_seq(k_f32, h_k, s_kv_i, d_qk, torch_itype, block_size, columnwise=False)
+        v8, dqv, sfv = _quantize_seq(v_f32, h_v, s_kv_i, d_vo, torch_itype, block_size, columnwise=True)
+        q8_seqs.append(q8); k8_seqs.append(k8); v8_seqs.append(v8)
```

Ruff (0.16.3)

[error] Multiple statements on one line (semicolon)

(E702)